cmd/aries-agent-lambda/aries.go

```diff
@@ -0,0 +1,387 @@
+	msgHandler                                     command.MessageHandler
+	dbParam                                        *dbParam
+	autoExecuteRFC0593                             bool
+}
+
+const (
+	httpProtocol      = "http"
+	websocketProtocol = "ws"
+	defaultWebhookUrl = "https://sly.slygo.scaffoldly.dev/aries-webhooks/api/v1/webhooks"
+)
+
+var (
+	keyTypes = map[string]kms.KeyType{
+		"ed25519":           kms.ED25519Type,
+		"ecdsap256ieee1363": kms.ECDSAP256TypeIEEEP1363,
+		"ecdsap256der":      kms.ECDSAP256TypeDER,
+		"ecdsap384ieee1363": kms.ECDSAP384TypeIEEEP1363,
+		"ecdsap384der":      kms.ECDSAP384TypeDER,
+		"ecdsap521ieee1363": kms.ECDSAP521TypeIEEEP1363,
+		"ecdsap521der":      kms.ECDSAP521TypeDER,
+	}
+
+	keyAgreementTypes = map[string]kms.KeyType{
+		"x25519kw": kms.X25519ECDHKWType,
+		"p256kw":   kms.NISTP256ECDHKWType,
+		"p384kw":   kms.NISTP384ECDHKWType,
+		"p521kw":   kms.NISTP521ECDHKWType,
+	}
+)
+
+func RegisterRoutes(router *mux.Router) (*mux.Router, error) {
+	// Tidbits from https://github.com/hyperledger/aries-framework-go/blob/9446ee81c3b4f39ca9f7af4abcf7cb4cc594e968/cmd/aries-agent-rest/startcmd/start.go#L736
+	dbParam := &dbParam{
+		dbType: "dynamodb",
+	}
+
+	httpProto := "https"
+	wsProto := "wss"
+	stage := os.Getenv("STAGE")
+	httpHost := os.Getenv("API_GATEWAY_DOMAIN")
+	wsHost := os.Getenv("API_GATEWAY_WEBSOCKET_DOMAIN")
+	internalHttpHost := httpHost
+	// internalWsHost := wsHost
+	basePath := os.Getenv("SERVICE_SLUG")
+	webhookPath := fmt.Sprintf("/%s/%s", basePath, "webhook")
+	if stage == "local" {
+		httpProto = "http"
+		wsProto = "ws"
+		httpHost = "localhost:3000"
+		wsHost = "localhost:3001"
+		internalHttpHost = "host.docker.internal:3000"
+		// internalWsHost = "host.docker.internal:3001"
+	}
+	inboundHttpHost := fmt.Sprintf("http@%s", httpHost)
+	log.Printf("Inbound Http Host: %s", inboundHttpHost)
+	inboundWsHost := fmt.Sprintf("ws@%s", wsHost)
+	log.Printf("Inbound Ws Host: %s", inboundWsHost)
+	inboundHostHttpExternalUrl := fmt.Sprintf("http@%s://%s/%s", httpProto, httpHost, basePath)
+	log.Printf("Inbound Http Host External URL: %s", inboundHostHttpExternalUrl)
+	inboundHostWsExternalUrl := fmt.Sprintf("ws@%s://%s/%s", wsProto, wsHost, basePath)
+	log.Printf("Inbound Ws Host External URL: %s", inboundHostWsExternalUrl)
+	// TODO: Convert defaultWebhookUrl to Environment Variables
+	webhookUrls := []string{defaultWebhookUrl, fmt.Sprintf("%s://%s%s", httpProto, internalHttpHost, webhookPath)}
+	log.Printf("Webhook URLs: %s", webhookUrls)
+
+	parameters := &agentParameters{
+		// server:               server,
+		host: httpHost,
+		// token:                token,
+		inboundHostInternals: []string{inboundWsHost},
+		inboundHostExternals: []string{inboundHostWsExternalUrl},
+		dbParam:              dbParam,
+		// defaultLabel:         defaultLabel,
+		webhookURLs: webhookUrls,
+		// httpResolvers:        httpResolvers,
+		outboundTransports: []string{"ws"},
```

Comment:
We've noticed invitations returning the http path, if we had `[]string{"ws", "http"}` here.

Reply:
@rolsonquadras @sudeshrshetty @fqutishat

Reply:
@borancar Are you talking about paths in service endpoints of invitations?